sdk/servicebus/service-bus/test/internal/receiver.spec.ts

```diff
@@ -0,0 +1,249 @@
+      assert.isFalse(
+        closeWasCalled,
+        "sanity check, subscription.close() does not close the receiver"
+      );
+      assert.isTrue(
+        receiverWasDrained,
+        "receiver should drain when subscription.close() is called"
+      );
+
+      await receiverImpl.close();
+      // rhea receiver is finally closed when the overall Receiver class is closed.
+      assert.isTrue(closeWasCalled, "receiver should note that we closed");
+    });
+
+    it("can't subscribe while another subscribe is active", async () => {
+      const receiverImpl = new ReceiverImpl(createClientEntityContextForTests(), "peekLock");
+
+      const subscription = await subscribeAndWaitForInitialize(receiverImpl);
+
+      try {
+        await subscribeAndWaitForInitialize(receiverImpl);
+        assert.fail("Should throw since we have an active subscriber");
+      } catch (err) {
+        assert.deepEqual(
+          {
+            name: err.name,
+            message: err.message
+          },
+          {
+            name: "Error",
+            message: 'The receiver for "queue" is already receiving messages.'
+          }
+        );
+      }
+
+      await subscription.close();
+      await receiverImpl.close();
+    });
+
+    it("can re-subscribe after previous subscription is closed", async () => {
+      let closeWasCalled = false;
+
+      const receiverImpl = new ReceiverImpl(
+        createClientEntityContextForTests({
+          onCreateReceiverCalled: (receiver) => {
+            (receiver as any).close = () => {
+              closeWasCalled = true;
+            };
+          }
+        }),
+        "peekLock"
+      );
+
+      const subscription = await subscribeAndWaitForInitialize(receiverImpl);
+      const originalStreamingReceiver = receiverImpl["_context"].streamingReceiver;
+
+      await subscription.close();
+
+      assert.isFalse(
+        closeWasCalled,
+        "closing a subscription does NOT close the receiver (it should be re-usable)"
+      );
+
+      const subscription2 = await subscribeAndWaitForInitialize(receiverImpl);
+
+      assert.equal(
+        originalStreamingReceiver?.name,
+        receiverImpl["_context"].streamingReceiver?.name,
+        "StreamingReceiver is closed but not replaced - this allows us to just stop and start at will without losing anything."
+      );
+
+      await subscription2.close();
+
+      await receiverImpl.close();
+    });
+
+    it("can re-subscribe after previous subscription is aborted", async () => {
+      const receiverImpl = new ReceiverImpl(createClientEntityContextForTests(), "peekLock");
+
+      const abortSignal = {
+        aborted: true
+      } as AbortSignalLike;
+
+      const subscription = receiverImpl.subscribe(
+        {
+          processError: async (_err) => {},
+          processMessage: async (_msg) => {}
+        },
+        {
+          //
+          // TODO: why is tracingOptions mandatory now?
+          //
+          tracingOptions: {},
```

Comment:
No, it is not. You can remove this.

Reply:
![image](https://user-images.githubusercontent.com/10452642/89588321-4227fe80-d7f8-11ea-83ae-4270d93882f8.png)

Reply:
Fixed. It's interesting - I was getting a compile error but perhaps it was just a momentary glitch. :)